README.md

```diff
@@ -23,396 +23,373 @@ Stable release versions are manually released from the main branch.
-signature set, and propagates it to Bitcoin.
+> **Note**: This method focuses on creating Babylon chain transactions, 
+> not Bitcoin transactions. Specifically:
+>
+> It creates Babylon transactions that contain:
+>   - Unsigned Staking transaction
+>   - Unsigned Unbonding Bitcoin transaction
+>   - Signed Slashing Bitcoin transactions
+>   - Proof of possession
 
-For completeness, we present the alternative method in which the application
-has access to the covenant signature set. This method can be useful for testing
-purposes. It involves the following:
 
 ```ts
-// Create the full witness
-const witness = createWitness(
-  unbondingTx.ins[0].witness: Buffer[], // original witness
-  covenantPks: Buffer[],
-  covenantUnbondingSignatures: {
-    btc_pk_hex: string;
-    sig_hex: string;
-  }[],
+import {
+  BabylonBtcStakingManager,
+} from "@babylonlabs-io/btc-staking-ts";
+
+const manager = new BabylonBtcStakingManager(
+  btcNetwork,
+  stakingParams,
+  btcProvider,
+  bbnProvider,
 );
 
-// Put the witness inside the unbonding transaction.
-unbondingTx.setWitness(0, witness);
+const { 
+  signedBabylonTx
+} = await manager.preStakeRegistrationBabylonTransaction(
+  stakerInfo,
+  stakingInput,
+  babylonBtcTipHeight,
+  inputUTXOs,
+  feeRate,
+  bech32Address,
+);
 ```
 
-### Withdrawing
+### Staking Transaction
 
-Withdrawing involves extracting funds for which the staking/unbonding period
-has expired from the staking/unbonding transaction.
+This step signs the previously created (but unsigned) Bitcoain staking transaction, 
+making it ready for submission to the Bitcoin network. The hash of the signed 
+staking transaction must match the staking transaction hash contained within 
+the pre-staking registration Babylon transaction.
 
-Initially, we specify the withdrawal transaction parameters.
+The unsigned staking transaction can be retrieved from either the Babylon
+chain or the backend API.
 
 ```ts
-// The index of the staking/unbonding output in the staking/unbonding
-// transcation.
-const stakingOutputIndex: number = 0;
-
-// The fee that the withdrawl transaction should use.
-const withdrawalFee: number = 500;
+import {
+  BabylonBtcStakingManager,
+} from "@babylonlabs-io/btc-staking-ts";
+
+const manager = new BabylonBtcStakingManager(
+  btcNetwork,
+  stakingParams,
+  btcProvider,
+  bbnProvider,
+);
 
-// The address to which the funds should be withdrawed to.
-const withdrawalAddress: string = btcWallet.address;
+const signedBtcStakingTx = await manager.createSignedBtcStakingTransaction({
+  stakerInfo,
+  stakingInput,
+  unsignedStakingTx,
+  inputUTXOs,
+  stakingParamsVersion
+})
 ```
 
-Then, we construct the withdrawal transaction.
-There are two types of withdrawal
+### Unbonding Transaction
 
-1. Withdraw funds from a staking transaction in which the timelock naturally
-   expired:
+This step signs the previously created unbonding transaction and 
```

Comment:
Are we following a sequence with this instructions (referring to the "this step")? Also, what's the previously created unbonding transaction?